subiquity/server/controllers/filesystem.py

```diff
@@ -638,8 +646,9 @@ async def guided_core_boot(self, disk: Disk):
             else:
                 if structure.role == snapdapi.Role.SYSTEM_DATA and \
                    structure == self._on_volume.structure[-1]:
-                    gap = gaps.largest_gap(disk)
+                    gap = gaps.at_offset(disk, offset)
                     size = gap.size - (offset - gap.offset)
+                    assert size > 0
```

Comment:
These changes don't seem quite the same thing as the others in this branch.

Reply:
Fair.  Moved to https://github.com/canonical/subiquity/pull/1674